packages/amplify-provider-awscloudformation/src/download-api-models.ts

```diff
@@ -29,41 +30,27 @@ function downloadAPIModels(context, allResources) {
   return sequential(promises);
 }
 
-function extractAPIModel(context, resource, framework) {
-  return new APIGateway(context).then(apigw => {
-    const apigwParams = getAPIGWRequestParams(context, resource, framework);
-
-    const apiName = resource.output.ApiName;
-
-    return apigw.apigw
-      .getSdk(apigwParams)
-      .promise()
-      .then(data => {
-        const backendDir = context.amplify.pathManager.getBackendDirPath();
-
-        const tempDir = `${backendDir}/.temp`;
-
-        fs.ensureDirSync(tempDir);
-
-        const buff = Buffer.from(data.body);
-        return new Promise((resolve, reject) => {
-          fs.writeFile(`${tempDir}/${apiName}.zip`, buff, err => {
-            if (err) {
-              reject(err);
-            }
-            extract(`${tempDir}/${apiName}.zip`, { dir: tempDir }, err => {
-              if (err) {
-                reject(err);
-              }
-              // Copy files to src
-              copyFilesToSrc(context, apiName, framework);
-              fs.removeSync(tempDir);
-              resolve();
-            });
-          });
-        });
-      });
-  });
+async function extractAPIModel(context: $TSContext, resource, framework) {
+  const apigw = await APIGateway.getInstance(context);
+  const apigwParams = getAPIGWRequestParams(context, resource, framework);
+
+  const apiName = resource.output.ApiName;
+
+  const data = await apigw.apigw.getSdk(apigwParams).promise();
```

Comment:
Shouldn't it be the following?
```suggestion
  const data = await apigw.getSdk(apigwParams).promise();
```

Reply:
APIGW class is just an wrapper which has instance of actual class. The only reason I changed the wrapper class was not create an async constructor of the class